core/dbt/contracts/graph/model_config.py

```diff
@@ -658,9 +663,17 @@ def finalize_and_validate(self):
 BASE_RESOURCE_TYPES.update({NodeType.Snapshot: EmptySnapshotConfig})
 
 
-def get_config_for(resource_type: NodeType, base=False) -> Type[BaseConfig]:
+def get_config_for(
+    resource_type: NodeType, base=False, language: Optional[ModelLanguage] = None
+) -> Type[BaseConfig]:
     if base:
         lookup = BASE_RESOURCE_TYPES
     else:
         lookup = RESOURCE_TYPES
-    return lookup.get(resource_type, NodeConfig)
+
+    resource = lookup.get(resource_type, NodeConfig)
+
+    if language == "python" and (resource == NodeConfig):
```

Comment:
Let's use the enum value here instead of a string literal.

Reply:
Here's the enum:

https://github.com/dbt-labs/dbt-core/blob/a7af3b383139851d4ad6f4e3859cdd961330a384/core/dbt/artifacts/resources/types.py#L54-L56

Here's an example of using it:

https://github.com/dbt-labs/dbt-core/blob/a7af3b383139851d4ad6f4e3859cdd961330a384/core/dbt/parser/base.py#L25

https://github.com/dbt-labs/dbt-core/blob/a7af3b383139851d4ad6f4e3859cdd961330a384/core/dbt/parser/base.py#L218